File: backend/src/main/java/lan/dk/podcastserver/service/properties/ExternalTools.java
package lan.dk.podcastserver.service.properties;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Created by kevin on 12/04/2016 for Podcast Server
 */
@Getter @Setter
@Accessors(chain = true)
@ConfigurationProperties("podcastserver.externaltools")
public class ExternalTools {
    private String ffmpeg = "/usr/local/bin/ffmpeg";
    private String rtmpdump = "/usr/local/bin/rtmpdump";
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/service/properties/ExternalTools.kt
package com.github.davinkevin.podcastserver.service.properties

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * Created by kevin on 12/04/2016 for Podcast Server
 */
@ConfigurationProperties("podcastserver.externaltools")
data class ExternalTools(
        val ffmpeg: String = "/usr/local/bin/ffmpeg",
        val ffprobe: String = "/usr/local/bin/ffprobe",
        val rtmpdump: String = "/usr/local/bin/rtmpdump",
        val youtubedl: String = "/usr/local/bin/youtube-dl"
)
